server/store/store.go

```diff
@@ -68,33 +70,48 @@ func (s *Store) Delete(key string) error {
 	return nil
 }
 
-func (s *Store) DeleteUserTokenOnEncryptionSecretChange() error {
+func (s *Store) DeleteUserTokenOnEncryptionSecretChange() {
 	page := 0
+	wg := new(sync.WaitGroup)
+
 	for {
 		kvList, err := s.api.KVList(page, constants.UsersPerPage)
 		if err != nil {
-			return err
-		}
-		if len(kvList) == 0 {
-			return nil
+			s.api.LogError("Failed to get the KVList", "Error", err.Error())
+			return
 		}
 
 		// isUserDeleted flag is used to check the condition for increasing the page number.
 		// If a key is deleted, the keys present in the list after the deleted key fills up the index of the deleted key by taking that position.
 		// If a key is deleted we don't increase the page number, else we increase it by 1.
 		isUserDeleted := false
 		for _, key := range kvList {
-			if userID, isValidUserKey := IsValidUserKey(key); isValidUserKey {
-				isUserDeleted = true
-				if err := s.Delete(key); err != nil {
-					return err
+			wg.Add(1)
+
+			go func(key string) {
+				defer wg.Done()
+
+				if userID, isValidUserKey := IsValidUserKey(key); isValidUserKey {
+					isUserDeleted = true
+					if isDeleted, err := s.DeleteUser(userID); !isDeleted {
```

Comment:
What's the need for `isUserDeleted` return value? We can just use `error` as the return type.

Reply:
Sure